slick/src/main/scala/slick/jdbc/JdbcBackend.scala

```diff
@@ -287,7 +294,7 @@ trait JdbcBackend extends RelationalBackend {
       val source = JdbcDataSource.forConfig(usedConfig, driver, path, classLoader)
       val poolName = usedConfig.getStringOr("poolName", path)
       val numThreads = usedConfig.getIntOr("numThreads", 20)
-      val maxConnections = source.maxConnections.fold(numThreads*5)(identity)
+      val maxConnections = source.maxConnections.getOrElse(numThreads*5)
```

Comment:
Changed in #1855 to be `numThreads`, without `* 5` because having more connections than threads means a connection without a thread can be holding a lock on the db, preventing all of the threads from continuing.

Reply:
Ok, correct. 
I'm however updating this line. A `fold` on an `Option` using `identity` is a overkill and also obscure. `getOrElse` should be preferred.